BangazonWorkForceManagement/Controllers/EmployeesController.cs

```diff
@@ -0,0 +1,238 @@
+        {
+            _config = config;
+        }
+
+        public SqlConnection Connection
+        {
+            get
+            {
+                return new SqlConnection(_config.GetConnectionString("DefaultConnection"));
+            }
+        }
+
+        // GET: Employees
+        public ActionResult Index()
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"SELECT e.FirstName AS EmployeeFirstName,
+                                               e.Id AS EmployeeId,
+                                               e.IsSupervisor AS IsSupervisor,
+	                                           e.LastName AS EmployeeLastName,
+                                               d.Budget AS DepartmentBudget,
+	                                           d.Name AS DepartmentName,
+                                               d.Id as DepartmentId
+                                        FROM Employee e
+                                        JOIN Department AS d on d.Id = e.DepartmentId";
+                    SqlDataReader reader = cmd.ExecuteReader();
+
+                    List<Employee> employees = new List<Employee>();
+                    while (reader.Read())
+                    {
+                        Employee employee = new Employee
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("EmployeeId")),
+                            FirstName = reader.GetString(reader.GetOrdinal("EmployeeFirstName")),
+                            LastName = reader.GetString(reader.GetOrdinal("EmployeeLastName")),
+                            IsSuperVisor = reader.GetBoolean(reader.GetOrdinal("IsSupervisor")),
+                            DepartmentId = reader.GetInt32(reader.GetOrdinal("DepartmentId")),
+                            Department = new Department
+                            {  
+                                Id = reader.GetInt32(reader.GetOrdinal("DepartmentId")),
+                                Name = reader.GetString(reader.GetOrdinal("DepartmentName")),
+                                Budget = reader.GetInt32(reader.GetOrdinal("DepartmentBudget"))
+                            }                        
+                        };
+
+                        employees.Add(employee);
+                    }
+                    reader.Close();
+                    return View(employees);
+                }
+            }
+        }
+
+        // GET: Employees/Details/5
+        public ActionResult Details(int id)
+        {
+            using (SqlConnection conn = Connection)
+
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @" 
+										SELECT e.FirstName AS EmployeeFirstName,
```

Comment:
I *thought* I'd left this comment earlier, but now I don't see it. I'm sorry.

How do you know that your query will return the employee's current computer?

Reply:
Corrected!